src/app/battle.service.ts

```diff
@@ -49,9 +49,9 @@ export class BattleService {
       design.forEach(ds => {
         const ship = new Ship();
```

Comment:
I didn't went with the object pool method of creating new ship since the modules property would just make things ether messy/ugly or overly complicated.

1 ship object requiring GC per fight is small enough to be a non-issue compare to over 10000 ship object per fight.